.github/actions/spelling/action.yml

```diff
@@ -3,56 +3,148 @@ name: Spelling
+        (Resolve-Path "${{ github.action_path }}/../../../Scripts").Path >> $Env:GITHUB_PATH
+    
+    # If the configuration path doesn't exist in the workspace, then it means that the workspace repository doesn't have
+    # its own spell-checking configuration. In that case we don't need to copy anything, because the configuration files
+    # will be used implicitly according to the value of default-configuration-repository.
+    - name: Check configuration path
+      id: check-configuration-path
+      shell: pwsh
+      run: |
+        Set-GitHubOutput -Key 'configuration-path-exists' -Value (Test-Path (Join-Path "$Env:GITHUB_WORKSPACE" '${{ inputs.configuration-path }}'))
+    
+    - name: Copy spell-checking advice
+      if: steps.check-configuration-path.outputs.configuration-path-exists == 'true'
+      shell: pwsh
+      run: |
+        $targetPath = Join-Path "$Env:GITHUB_WORKSPACE" '${{ inputs.configuration-path }}' 'advice.md'
+
+        if (-not (Test-Path $targetPath))
+        {
+          Copy-Item (Join-Path '${{ github.action_path }}' 'advice.md') $targetPath
+        }
+
+    - name: Merge "excludes.txt"
+      if: inputs.merge-file-excludes == 'true' && steps.check-configuration-path.outputs.configuration-path-exists == 'true'
+      shell: pwsh
+      run: |
+        $mergeParameters = @{
+          Source = Join-Path '${{ github.action_path }}' 'excludes.txt'
+          Target = Join-Path "$Env:GITHUB_WORKSPACE" '${{ inputs.configuration-path }}' 'excludes.txt'
+        }
+
+        Merge-SpellCheckingDictionaryFile @mergeParameters
+
+    - name: Merge "line_forbidden.patterns"
+      if: inputs.merge-forbidden-patterns == 'true' && steps.check-configuration-path.outputs.configuration-path-exists == 'true'
+      shell: pwsh
+      run: |
+        $mergeParameters = @{
+          Source = Join-Path '${{ github.action_path }}' 'line_forbidden.patterns'
+          Target = Join-Path "$Env:GITHUB_WORKSPACE" '${{ inputs.configuration-path }}' 'line_forbidden.patterns'
+        }
+
+        Merge-SpellCheckingDictionaryFile @mergeParameters
+
+    - name: Merge "patterns.txt"
+      if: inputs.merge-ignore-patterns == 'true' && steps.check-configuration-path.outputs.configuration-path-exists == 'true'
+      shell: pwsh
+      run: |
+        $mergeParameters = @{
+          Source = Join-Path '${{ github.action_path }}' 'patterns.txt'
+          Target = Join-Path "$Env:GITHUB_WORKSPACE" '${{ inputs.configuration-path }}' 'patterns.txt'
+        }
+
+        Merge-SpellCheckingDictionaryFile @mergeParameters
+
+    - name: Merge dictionary source prefixes
+      id: merge-dictionary-source-prefixes
+      shell: pwsh
+      run: |
+        $defaultPrefixes = @{
+          'lombiq-lgha' = 'https://raw.githubusercontent.com/Lombiq/GitHub-Actions/issue/OSOE-523/.github/actions/spelling/dictionaries/'
+          'cspell' = 'https://raw.githubusercontent.com/check-spelling/cspell-dicts/v20220816/dictionaries/'
```

Comment:
Comment when and how these need to be updated.

Reply:
'lombiq-lgha' will be updated to use the dev branch before merging this PR and 'cspell' has the associated instructions in the actual spelling step below, because that's where the spelling version is updated (and that's what you need to start with).

On a related note though, I'll expand the docs with instructions for using a branch of LGHA other than dev, because the value of 'lombiq-lgha' should be overridden (so that the correct version of e.g., Lombiq.common.txt) is applied.